piece.py

```diff
@@ -1,14 +1,12 @@
-class Piece(object):
-    def __init__(self, name, team, piece_type):
+class Piece (object):
+
+    def __init__(self, name, team, piece_type, pos):
         self.name = name
         self.team = team
         self.type = piece_type
         self.his_pos = []
-        self.pass_river = False
-        self.nick_name = team[0] + '_' + self.type[0:3] + '_' + self.name[-1]
-
-    def set_init_pos(self, pos):
         self.his_pos.append(pos)
```

Comment:
- Position should not be declare directly in init(). This is because when the piece is first initialized, it does not have the position yet. We have to initialize the position of the piece later so the position will add an unnecessary position of (0, 0) to the his_pos list. Refer to the funtions: init_dict(), generate_init_pos() and init_piece_pos() to be clear.
- Also, I think would be good to use.

Reply:
This is the copy from your Piece class. Prob the old version